lib/routes/osu/beatmaps/latest-ranked.ts

```diff
@@ -0,0 +1,349 @@
+    user_id: number;
+    version: string;
+    accuracy: number;
+    ar: number;
+    bpm: number;
+    convert: boolean;
+    count_circles: number;
+    count_sliders: number;
+    count_spinners: number;
+    cs: number;
+    deleted_at: string | null;
+    drain: number;
+    hit_length: number;
+    is_scoreable: boolean;
+    last_updated: string;
+    mode_int: number;
+    passcount: number;
+    playcount: number;
+    ranked: number;
+    url: string;
+    checksum: string;
+    max_combo: number;
+}
+
+interface NominationsSummary {
+    current: number;
+    eligible_main_rulesets: string[];
+    required_meta: {
+        main_ruleset: number;
+        non_main_ruleset: number;
+    };
+}
+
+interface Covers {
+    cover: string;
+    'cover@2x': string;
+    card: string;
+    'card@2x': string;
+    list: string;
+    'list@2x': string;
+    slimcover: string;
+    'slimcover@2x': string;
+}
+
+interface Availability {
+    download_disabled: boolean;
+    more_information: string | null;
+}
+
+interface BeatmapsetInfo {
+    artist: string;
+    artist_unicode: string;
+    covers: Covers;
+    creator: string;
+    favourite_count: number;
+    hype: null | string;
+    id: number;
+    nsfw: boolean;
+    offset: number;
+    play_count: number;
+    preview_url: string;
+    source: string;
+    spotlight: boolean;
+    status: string;
+    title: string;
+    title_unicode: string;
+    track_id: number;
+    user_id: number;
+    video: boolean;
+    bpm: number;
+    can_be_hyped: boolean;
+    deleted_at: string | null;
+    discussion_enabled: boolean;
+    discussion_locked: boolean;
+    is_scoreable: boolean;
+    last_updated: string;
+    legacy_thread_url: string;
+    nominations_summary: NominationsSummary;
+    ranked: number;
+    ranked_date: string;
+    storyboard: boolean;
+    submitted_date: string;
+    tags: string;
+    availability: Availability;
+    beatmaps: Beatmap[];
+    pack_tags: string[];
+}
+
+async function handler(ctx): Promise<Data> {
+    // Parse & retrive searchParams
+    const pathParams = ctx.req.param('routeParams');
+    // Here user actually pass the query using path param, like: `/osu/latest-ranked/includeMode=osu`
+    // We first retrieve path param part: `includeMode=osu`, then concat it with host to construct a "fake" URL:
+    // `https://osu.ppy.sh?includeMode=osu`
+    // Then we use URL.searchParams to parse and retrieve params from this "fake" URL.
+    const searchParams = new URL(`https://osu.ppy.sh?${pathParams}`).searchParams; // use URL to parse params
+    const includeModes = searchParams.getAll('includeMode');
+    const difficultyLimits = searchParams.getAll('difficultyLimit');
+    const modeInTitle = searchParams.get('modeInTitle') ?? 'true'; // show mode name in title, default to true.
+
+    // fetch beatmap JSON info from website within cache
+    let beatmapsetList = (await cache.tryGet('https://osu.ppy.sh/beatmapsets:JSON', async () => {
```

Comment:
Please set the 4th parameter of `tryGet` to `false` and 3rd to `config.cache.routeExpire`, otherwise the [cache won't be updated](https://docs.rsshub.app/joinus/advanced/use-cache#parameters).